### PR TITLE
bugfix: fix pod-like components naming with "component" word in name

### PR DESCRIPTION
**Pull request by lifeart**

### CODE REVIEW

File: src/utils/path-matcher.ts [hunk 1]
```diff
@@ -140,7 +140,11 @@ export class PodMatcher extends ClassicPathMatcher {
     serializer: ['/serializer.'],
   };
   rightPartFromFirstMatch(_: string, fileName: string, extName: string, str: string) {
-    const fullName = str.slice(str.indexOf(this.podPrefix) + this.podPrefix.length + 1, str.length).slice(0, -(1 + extName.length + fileName.length));
+    let fullName = str.slice(str.indexOf(this.podPrefix) + this.podPrefix.length + 1, str.length).slice(0, -(1 + extName.length + fileName.length));
```
Comment (NullVoxPopuli) [inline src/utils/path-matcher.ts]: should this be a method?

Comment (NullVoxPopuli) [reply src/utils/path-matcher.ts]: maybe with some variables split out? ;)

File: test/utils/path-matcher-test.ts [hunk 2]
```diff
@@ -201,6 +213,43 @@ describe('ClassicPathMatcher', () => {
...
+      scope: 'application',
+      kind: 'script',
+    });
+    expect(m('repos/els-addon-typed-templates/app/components/my-component/template.hbs')).toEqual({
+      type: 'component',
+      name: 'my-component',
+      scope: 'application',
+      kind: 'template',
+    });
+    expect(m('repos/els-addon-typed-templates/app/components/my-component/index.ts')).toEqual({
```
Comment (NullVoxPopuli) [inline test/utils/path-matcher-test.ts:229]: ah! this makes sense. I have a lot of these

File: src/utils/path-matcher.ts (hunk 1 above)
Comment (lifeart) [reply src/utils/path-matcher.ts]: @NullVoxPopuli good catch! Changes landed

**Review by NullVoxPopuli — APPROVED** [review]

**Pull request merged**